modules/samples/firealarm/component/plugin/src/main/java/org/homeautomation/firealarm/plugin/impl/dao/impl/DeviceTypeDAOImpl.java

```diff
@@ -0,0 +1,239 @@
+            stmt = conn.prepareStatement(deleteDBQuery);
+            stmt.setString(1, deviceId);
+            int rows = stmt.executeUpdate();
+            if (rows > 0) {
+                status = true;
+                if (log.isDebugEnabled()) {
+                    log.debug("Connected Cup device " + deviceId + " data has deleted" +
+                              " from the Connected Cup database.");
+                }
+            }
+        } catch (SQLException e) {
+            String msg = "Error occurred while deleting Connected Cup device " + deviceId;
+            log.error(msg, e);
+            throw new FirealarmPluginException(msg, e);
+        } finally {
+            DeviceTypeUtils.cleanupResources(stmt, null);
+        }
+        return status;
+    }
+
+    public List<Device> getAllDevices() throws FirealarmPluginException {
+        Connection conn;
+        PreparedStatement stmt = null;
+        ResultSet resultSet = null;
+        Device connectedCupDevice;
+        List<Device> iotDevices = new ArrayList<>();
+
+        try {
+            conn = DeviceTypeDAO.getConnection();
+            String selectDBQuery =
+                    "SELECT firealarm_DEVICE_ID, DEVICE_NAME, ACCESS_TOKEN, REFRESH_TOKEN" +
+                    "FROM firealarm_DEVICE";
+            stmt = conn.prepareStatement(selectDBQuery);
+            resultSet = stmt.executeQuery();
+            while (resultSet.next()) {
+                connectedCupDevice = new Device();
+                connectedCupDevice.setDeviceIdentifier(resultSet.getString(
+                        DeviceTypeConstants.DEVICE_PLUGIN_DEVICE_ID));
+                connectedCupDevice.setName(resultSet.getString(
+                        DeviceTypeConstants.DEVICE_PLUGIN_DEVICE_NAME));
+
+                List<Device.Property> propertyList = new ArrayList<>();
+                propertyList.add(DeviceTypeUtils.getProperty(
+                        DeviceTypeConstants.DEVICE_PLUGIN_PROPERTY_ACCESS_TOKEN,
+                        resultSet.getString("ACCESS_TOKEN")));
+                propertyList.add(DeviceTypeUtils.getProperty(
+                        DeviceTypeConstants.DEVICE_PLUGIN_PROPERTY_REFRESH_TOKEN,
+                        resultSet.getString("REFRESH_TOKEN")));
+                connectedCupDevice.setProperties(propertyList);
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("All Connected Cup device details have fetched from Connected Cup database" +
+                          ".");
+            }
+            return iotDevices;
+        } catch (SQLException e) {
+            String msg = "Error occurred while fetching all Connected Cup device data'";
+            log.error(msg, e);
+            throw new FirealarmPluginException(msg, e);
+        } finally {
+            DeviceTypeUtils.cleanupResources(stmt, resultSet);
+            DeviceTypeDAO.closeConnection();
```

Comment:
normally a .closeConnection() should occur at the business logic layer, not at the DAO layer.
just cleaning up the resources would be adequate here.